pkg/frontend/querymiddleware/codec.go

```diff
@@ -311,12 +313,18 @@ func (prometheusCodec) DecodeLabelsQueryRequest(_ context.Context, r *http.Reque
 
 	labelMatcherSets := reqValues["match[]"]
 
+	limit, err := strconv.ParseInt(reqValues.Get("limit"), 10, 64)
```

Comment:
Any reason to not use `ParseUint()` here since a limit of less than 0 doesn't really make sense? Does Prometheus accept negative values?

Reply:
> Any reason to not use ParseUint() here since a limit of less than 0 doesn't really make sense?

no reason. I changed the unit of the `limit` to `uint64` throughout

> Does Prometheus accept negative values?

no it doesn't https://github.com/grafana/mimir/blob/7408575db4f833f7561935d6252859d8c5663b73/vendor/github.com/prometheus/prometheus/web/api/v1/api.go#L1898-L1913